src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -743,7 +743,7 @@ public void ProcessBlock(Block block, ChainedBlock chainedBlock)
         }
 
         /// <inheritdoc />
-        public void ProcessTransaction(Transaction transaction, int? blockHeight = null, Block block = null)
+        public void ProcessTransaction(Transaction transaction, int? blockHeight = null, Block block = null, bool? isPropagated = null)
```

Comment:
I would make this non-null and `true` by default (`bool isPropagated = true`).
Most of the transactions going through this method will come from blocks, where `isPropagated == true`. Then they will mostly come from other people's transactions appearing on the network, where `isPropagated == true`.
Only the transactions that we create will be `false` at first, then `true`.

This way, you only actively set isPropagated in one place, and it's to `false`, when you send a transaction.

Reply:
Actually here, instead of `true` by default, you can have null by default, just like here, and you can set false manually when necessary.
Then you make the isPropagated field `bool?` in TransactionData. This way, if it's false, we can see it in the object, if it's propagated, the field disappears (as it's not really needed anymore).

Reply:
So in this case not having the IsPropagated flag will mean that it is propagated... 

Hmm. From a logical perspective I'd rather have IsNotPropagated that will be null and removed from the serialized file as soon as it's propagated. 
Are you ok with that?

Reply:
And on top we'd have a method IsPropagated in TransactionData that will check if .IsNotPropagated null or not and return true if it is.

Reply:
But I'd actually just leave IsPropagated there and not delete it after it's true. 
Removing it will make it a bit more difficult to understand the code for other devs.